src/legacy/core_plugins/kibana/public/dashboard/lib/filter_utils.ts

```diff
@@ -19,7 +19,8 @@
 
 import _ from 'lodash';
 import moment, { Moment } from 'moment';
-import { Filter } from 'ui/embeddable';
+import { QueryFilter } from 'ui/filter_manager/query_filter';
+import { Filter } from '@kbn/es-query';
```

Comment:
TIL that `@kbn/es-query` _also_ has a filter interface!

So that interface is duplicated 3 places so far: ui/embeddables, ui/visualize, @kbn/es-query 

I agree the es-query implementation is the most thorough, so I guess we should decide eventually if the types stay there long term, or move to a plugin (perhaps data). Or maybe we just re-export them from data, who knows.

Reply:
Also embeddables and visualize export a `Filters` type which is equivalent to `Filter[]`; something we could consider adding to es-query (though TBH not sure how useful it really is).

Reply:
yea, I say nix the Filters type, seems easier to read to know that it's an array of Filter type and that is almost just as short to write.  @lizozom has a PR that starts this consolidation work (https://github.com/elastic/kibana/pull/37311) but was halted for the time being due to it starting to conflict with the changing coming with the embeddable stuff.